Improve denon media_player

glance- opened the pull request:
This reworks and updates the denon media player component.

This is so it can better represent the Denon recivers who also talks this protocol.

This have bin tested against a Denon AVR 1912.

glance- committed:
Added media stop for Denon AVR Media Player

glance- committed:
Rework input selection for Denon AVR

This reworks the input selection, adding more modes and making it so
that the media controls are only announced in modes where they actually
makes sense.

glance- committed:
Added real media info for Denon AVR Media modes

glance- committed:
Sort source list

glance- committed:
Add debug level logging of messages in denon

MartinHjelmare commented on homeassistant/components/media_player/denon.py:
```diff
@@ -122,17 +152,28 @@ def is_volume_muted(self):
+    @property
+    def source(self):
+        """Return the current input source."""
+        for pretty_name, name in self._source_list.items():
```
To return the first match only, you could do something like this:
```py
next((k for k, v in input_dict.items() if v == value), None)
```

glance- replied:
I'm not really sure if i think that makes the code easier to read, but if that's how the project likes it i'm fine with that.

MartinHjelmare replied:
Generator expressions and iterators are standard python, so I think it's good to use those when there's a use case, and it will use less lines.

balloob replied:
I have recently actually started to prefer the `for key, val in x:` approach for readability. But both are ok .

MartinHjelmare replied:
In my opinion, the python syntax for generator expressions and list or dict comprehensions is so good that you can understand them at first encounter. It's one elegant line that is not more convoluted than what would be written in three lines.

glance- committed:
Read more configuration from denon devices

This reads network name, and overrides the local name with that.

This also reads the source names and reconfigures the input list to
those names, and also reads the source deleted list and removes the
inputs that are set to deleted in the device.

glance- committed:
Discover and handle max volume in Denon media player

glance- committed:
Rework source discovery in Denon media player

This uses SSFUN as authorative source for which sources that we should
present.

balloob reviewed (approved):
It looks fine.

Someone should extract all the Telnet logic into its own lib to control Denon. This is too much protocol specific knowledge.
